API pages cleanup (#7224)
circleci/circleci-docs: jekyll/_cci2/api-developers-guide.md

of API endpoints will return JSON by default, but some endpoints (primarily API
v1) return EDN if no accept header is specified.

- To return formatted JSON, include a `text/plain` header like the example shown below:

  ```shell
  curl --header "Circle-Token: $CIRCLE_TOKEN" \
    --header 'Accept: text/plain'    \
    https://circleci.com/api/v2/project/{project-slug}/pipeline
  ```

- To return compressed JSON:

  ```shell
  curl --header "Circle-Token: $CIRCLE_TOKEN" \
    --header 'Accept: application/json'    \
    https://circleci.com/api/v2/project/{project-slug}/pipeline
  ```

## Getting started with the API
{: #getting-started-with-the-api }

> The CircleCI API shares similarities with previous API versions in that it identifies your projects using repository name. For instance, if you want to pull information from CircleCI about the GitHub repository "https://github.com/CircleCI-Public/circleci-cli" you can refer to that in the CircleCI API as `gh/CircleCI-Public/circleci-cli`, which is a "triplet" of the project type (VCS provider), the name of your "organization" (or your username), and the name of the repository.

Reviewer: We don't use quotes very often and I'm unsure if they fit here either. Maybe change "organization" to be more specific about the role and put it in bold like we do elsewhere when we reference things about roles? I'm also not sure the link belongs in a quote. Will the link still be clickable in quotes? If so, I think the quotes will make it seem more crowded. If the quotes escape the link and it's not clickable, then I do think they make sense in that instance.

    -d '{ "branch": "my-branch" }'
    ```

> 6. Let's move on to a more complex example: triggering a pipeline and passing a parameter that can be dynamically substituted into your configuration. In this example, we will pass a Docker image tag to our `docker` executor key. 

Reviewer: ```suggestion
6. Let us move on to a more complex example: triggering a pipeline and passing a parameter that can be dynamically substituted into your configuration. In this example, we will pass a Docker image tag to our `docker` executor key. 
```

    <br>
    First, we will need to modify the `.circleci/config.yml` to be a little more complex than the standard "Hello World" sample provided by the onboarding.

      ```yaml
      version: 2.1
      jobs:
        build:
          docker:
            - image: "circleci/node:<< pipeline.parameters.image-tag >>"
              auth:
                username: mydockerhub-user
                password: $DOCKERHUB_PASSWORD  # context / project UI env-var reference
          environment:
            IMAGETAG: "<< pipeline.parameters.image-tag >>"
          steps:
            - run: echo "Image tag used was ${IMAGETAG}"
      parameters:
        image-tag:
          default: latest
          type: string
      ```

>       You will need to declare the parameters you expect to receive from the API. In this case, under the `parameters` key, we define an "image-tag" to be expected in the JSON payload of a POST request to the _Trigger New Pipeline_ endpoint.

Reviewer: ```suggestion    
      You will need to declare the parameters you expect to receive from the API. In this case, under the `parameters` key, we define an `image-tag` to be expected in the JSON payload of a POST request to the [Trigger a new pipeline](https://circleci.com/docs/api/v2/index.html#operation/triggerPipeline) endpoint.
```

        type: string
    ```

>     You will need to declare the parameters you expect to receive from the API. In this case, under the `parameters` key, we define an "image-tag" to be expected in the JSON payload of a POST request to the _Trigger New Pipeline_ endpoint.

Reviewer: ```suggestion
    You will need to declare the parameters you expect to receive from the API. In this case, under the `parameters` key, we define an `image-tag` to be expected in the JSON payload of a POST request to the [Trigger a new pipeline](https://circleci.com/docs/api/v2/index.html#operation/triggerPipeline) endpoint.
```

## Additional API use cases
{: #additional-api-use-cases }

> Now that you have a general understanding of how the CircleCI API v2 service works through an end-to-end API example request and walkthrough, let's look at a few common tasks and operations you may perform on a regular basis when using the API.

Reviewer: ```suggestion
Now that you have a general understanding of how the CircleCI API v2 service works through an end-to-end API example request and walkthrough, let us look at a few common tasks and operations you may perform on a regular basis when using the API.
```

Before trying any of the API calls in this section, make sure you have a personal API token and have been authenticated to make calls to the server.

### Get project details
{: #get-project-details }

You may often find it helpful to retrieve information about a specific project, including the name of the organization the project belongs to, the version control system (VCS) that hosts the project, and other details. The CircleCI API enables you to return this and other information by making a single GET request to the `project/{project-slug}` endpoint by passing the `project-slug` parameter.

> **Note** If you would like more detailed information about a project, or need a refresher on the specifics of a project, please refer to the CircleCI [Projects]({{site.baseurl}}/projects/) page.

Reviewer: Should this remain a bold note or be turned into an official note banner?

### Get job details
{: #get-job-details }

> Much like the Get Project Details API request described in the previous example, the Get Job Details API request enables you to return specific job information from the CircleCI API by making a single API request. 

Reviewer: The "Get project details" endpoint was either mistyped or has changed -- it's now called "Get a project".
```suggestion
Much like the Get a project API request described in the previous example, the [Get job details](https://circleci.com/docs/api/v2/index.html#operation/getJobDetails) API request enables you to return specific job information from the CircleCI API by making a single API request. 
```

### Get job details
{: #get-job-details }

Much like the Get a project API request described in the previous example, the [Get job details](https://circleci.com/docs/api/v2/index.html#operation/getJobDetails) API request enables you to return specific job information from the CircleCI API by making a single API request. 

Retrieving job information can be very useful when you want information about how your job performed, what resources were used (e.g. pipeline, executor type, etc.), and the time it took for the job to finish.

#### Steps
{: #steps }

> {:.no_toc}

Reviewer 2: Could you take out the no_tocs so all sections appear in the right hand nav?

- `total credits used` - The total number of credits that were used during the build.
- `windows_start & windows_end` - The time the build was initiated, and then completed.

> **Note** The above example only shows a few builds. When you run this command, you may receive up to 250 individual builds that you can review in much more detail.

Reviewer: Should we change this bold not to an official banner note?
